### PR TITLE
Implement deletionProtection for the claims

### PR DESCRIPTION
**Pull request by TheBigLee**
## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

**Comment by TheBigLee** [comment]
Passing E2E tests: https://github.com/vshn/component-appcat/actions/runs/10143941384/job/28046390380

File: component/component/main.jsonnet [hunk 1]
```diff
@@ -84,6 +84,7 @@ local ns = kube.Namespace(params.namespace) {
   metadata+: {
     labels+: {
       'openshift.io/cluster-monitoring': 'true',
+      'appuio.io/organization': 'vshn',
```
Comment (Kidswiss) [inline component/component/main.jsonnet:87]: What's the purpose of this?

File: component/tests/vshn.yml [hunk 2]
```diff
@@ -152,7 +152,7 @@ parameters:
                   appuio.io/node-class: "plus"
               note: "Will be scheduled on APPUiO Cloud plus nodes"
         minio:
-          enabled: false
+          enabled: true
```
Comment (Kidswiss) [inline component/tests/vshn.yml:155]: Can you reset this? Usually we use kindev's minio as it's wired into the ingress.

**Review by Kidswiss — CHANGES_REQUESTED** [review]
Some nits

File: component/component/main.jsonnet (hunk 1 above)
Comment (TheBigLee) [reply component/component/main.jsonnet:87]: That is required for the VSHNMinio instance we deploy on the Lab. Without an organization label, the quota check will fail.

File: component/tests/vshn.yml (hunk 2 above)
Comment (TheBigLee) [reply component/tests/vshn.yml]: Done

**Review by Kidswiss — APPROVED** [review]

**Pull request merged**